File: Assets/code/CPG.cs
using UnityEngine;
using System;


public static class CPG
{
    public static void Update(float[] CPGs, float dt)
    {
        float Ao = 100, Bo = 120, Co = 1.5f, Do = 2.7f;
        float a = 1, b = 1;
        float tau1o = 8, tau2o = 16, tau3o = 150, tau4o = 150;

        float u = 2, u2 = 2;

        CPGs[0] += (dt / tau1o) * (-a * CPGs[0] + 
            (Ao * Mathf.Pow(150 - Do * CPGs[1], 2)) / 
            (Mathf.Pow(Bo + b * CPGs[2], 2) + Mathf.Pow(150 - Do * CPGs[1], 2)));

        CPGs[1] += (dt / tau2o) * (-a * CPGs[1] + 
            (Ao * Mathf.Pow(150 - Do * CPGs[0], 2)) / 
            (Mathf.Pow(Bo + b * CPGs[3], 2) + Mathf.Pow(150 - Do * CPGs[0], 2)));

        CPGs[2] += (dt / tau3o) * (-a * CPGs[2] + Co * CPGs[0]);
        CPGs[3] += (dt / tau4o) * (-a * CPGs[3] + Co * CPGs[1]);

        CPGs[4] += (dt / tau4o) * (-a * CPGs[4] + 1.01f * (0.5f * CPGs[0] + 0.5f * CPGs[1]));
        CPGs[5] = Mathf.Clamp(CPGs[5] + (dt / tau4o) * (-a * CPGs[5] + a * CPGs[0] - CPGs[4]), -u, u);
        CPGs[6] = Mathf.Clamp(CPGs[6] + (dt / tau4o) * (-a * CPGs[6] + a * CPGs[1] - CPGs[4]), -u, u);

        CPGs[7] += (dt / tau4o) * (-a * CPGs[7] + 1.01f * (0.5f * CPGs[2] + 0.5f * CPGs[3]));
        CPGs[8] = Mathf.Clamp(CPGs[8] + (dt / (tau4o * 0.5f)) * (-a * CPGs[8] + 1.2f * (CPGs[2] - CPGs[7])), -u2, u2);
        CPGs[9] = Mathf.Clamp(CPGs[9] + (dt / (tau4o * 0.5f)) * (-a * CPGs[9] + 1.2f * (CPGs[3] - CPGs[7])), -u2, u2);
    }
}

File: documentation/ControlNeuronal/CPG.m


function [CPG1, CPG2, CPG3, CPG4,CPG5, CPG6, CPG7,CPG8, CPG9, CPG10 ] = ...
    CPG(CPG1, CPG2, CPG3, CPG4,CPG5, CPG6,CPG7,CPG8,CPG9,CPG10, dt, param)
    % Parámetros obtenidos de la estructura
    Ao = param.Ao; Bo = param.Bo; Co = param.Co; Do = param.Do;
    a = param.a; b = param.b;
    tau1o = param.tau1o; tau2o = param.tau2o; tau3o = param.tau3o; tau4o = param.tau4o;  

    u = 50;   %%%%%%%%%%%%% parametro importante para modificar la marcha
    u2 = 50;
    CPG1 = CPG1 + (dt / tau1o) * (-a * CPG1 + (Ao * (150 - Do * CPG2)^2) / ((Bo + b * CPG3)^2 + (150 - Do * CPG2)^2));
    CPG2 = CPG2 + (dt / tau2o) * (-a * CPG2 + (Ao * (150 - Do * CPG1)^2) / ((Bo + b * CPG4)^2 + (150 - Do * CPG1)^2));
    CPG3 = CPG3 + (dt / tau3o) * (-a * CPG3 + Co * CPG1);
    CPG4 = CPG4 + (dt / tau4o) * (-a * CPG4 + Co * CPG2);
    
    CPG5 = CPG5 + (dt / tau4o) * (-a * CPG5 + 1.01*((1/2) * CPG1+(1/2) * CPG2));
    CPG6 = min(max((CPG6 + (dt / tau4o) * (-a * CPG6 + a * CPG1 -CPG5)),-u),u); %%% x1
    CPG7 = min(max((CPG7 + (dt / tau4o) * (-a * CPG7 + a * CPG2 -CPG5)),-u),u); 
    
    CPG8 = CPG8 + (dt / tau4o) * (-a * CPG8 + 1.01*((1/2) * CPG3 + (1/2) * CPG4));
    CPG9 = min(max((CPG9 + (dt / tau4o*0.5) * (-a * CPG9 + 1.2 * (CPG3 -CPG8))),-u2),u2); %%%%y2
    CPG10 = min(max((CPG10 + (dt / tau4o*0.5) * (-a * CPG10 + 1.2 * (CPG4 -CPG8))),-u2),u2);

    
end